core/iwasm/common/wasm_runtime_common.c

```diff
@@ -863,6 +863,183 @@ wasm_runtime_call_wasm(WASMExecEnv *exec_env,
     return false;
 }
 
+uint32 parse_args_to_uint32_array(WASMType *type,
+                                uint32 num_args, wasm_val_t *args,
+                                uint32 *out_argv)
+{
+    int i, p;
+
+    for (i = 0, p = 0; i < num_args; i++) {
+        switch (args[i].kind) {
+            case WASM_I32:
+                *(int32 *)out_argv = args[i].of.i32;
+                out_argv++;
+                p++;
+                break;
+            case WASM_I64:
+                *(int64 *)out_argv = args[i].of.i64;
+                out_argv += 2;
+                p += 2;
+                break;
+            case WASM_F32:
+                *(float32 *)out_argv = args[i].of.f32;
+                out_argv++;
+                p++;
+                break;
+            case WASM_F64:
+                *(float64 *)out_argv = args[i].of.f64;
+                out_argv += 2;
+                p += 2;
+                break;
+        }
+    }
+    return p;
+}
+
+uint32 parse_uint32_array_to_results(WASMType *type,
+                                uint32 argc, uint32 *argv,
+                                wasm_val_t *out_results)
+{
+    int i, p;
+
+    for (i = 0, p = 0; i < type->result_count; i++) {
+        switch (type->types[type->param_count + i]) {
+            case VALUE_TYPE_I32:
+                out_results[i].kind = WASM_I32;
+                out_results[i].of.i32 = *(int32 *)argv;
+                argv++;
+                p++;
+                break;
+            case VALUE_TYPE_I64:
+                out_results[i].kind = WASM_I64;
+                out_results[i].of.i64 = *(int64 *)argv;
```

Comment:
Same as above, should not convert argv to int64 pointer directly, had better use union to copy it.